templates/components/form/form.jsx

```diff
@@ -25,30 +27,47 @@ var Form = React.createClass({
     return {
       username: null,
       password: null,
-      email: null
+      email: null,
+      checked: false,
+      dirty: {}
     };
   },
   buildFormElement: function(key, i) {
-    var id = Object.keys(this.props.fields[i]);
+    // we always expect this.props.fields[i] to be one object with one property.
+    var self = this;
+    var id = Object.keys(this.props.fields[i])[0];
```

Comment:
extract as a `var values = var placeholder = this.props.fields[i][id];` and then just use straight up `values.type`, `values.label` etc? This is way too much this.props access it feels like